Add new config for rewrites/redirects/headers/etc, deprecate routes (#1498)
zeit/docs: components/references-mdx/configuration/routes/routes.mdx

  custom URL path in <InlineCode>dest</InlineCode>.
</Caption>

```json
{
  "rewrites": [{ "source": "/product/:id", "destination": "/product" }]
}
```

<Caption>

>   The equivalent to the routes example above using{' '}

Reviewer: This sentence didn't read well for me.
```suggestion
  An equivalent example to the <InlineCode>routes</InlineCode> usage above, but instead using{' '}
```

  keeping the path in the new location.
</Caption>

```json
{
  "redirects": [
    {
      "source": "/posts/(.*)",
      "destination": "/blog/$1",
      "statusCode": 301
    }
  ]
}
```

<Caption>

>   The equivalent to the routes example above using{' '}

Reviewer: ```suggestion
  An equivalent example to the <InlineCode>routes</InlineCode> usage above, but instead using{' '}
```

In this example, handle filesystem expands to route `/about` and `/contact`.

When using `rewrites`, handle filesystem is assumed so this example can be simplified to the following:

```json
{
  "rewrites": [{ "source": "/:slug", "destination": "/blog" }]
}
```

<Caption>

>   The equivalent to the routes example above using{' '}

Reviewer: ```suggestion
  An equivalent example to the <InlineCode>routes</InlineCode> example above, but instead using{' '}
```

  filesystem routes.
</Caption>

```json
{
  "rewrites": [{ "source": "/.*", "destination": "/index.html" }]
}
```

<Caption>

>   The equivalent to the routes example above using{' '}

Reviewer: ```suggestion
  An equivalent example to the <InlineCode>routes</InlineCode> example above, but instead using{' '}
```

  <InlineCode>continue</InlineCode> to add global headers.
</Caption>

```json
{
  "headers": [
    {
      "source": "/.*",
      "headers": [
        {
          "key": "Cache-Control",
          "value": "max-age=3600"
        }
      ]
    },
    {
      "source": "/blog.*",
      "headers": [
        {
          "key": "Cache-Control",
          "value": "max-age=600"
        }
      ]
    }
  ],
  "rewrites": [{ "source": "/blog/:slug", "destination": "/post" }]
}
```

<Caption>

>   The equivalent to the routes example above using{' '}

Reviewer: ```suggestion
  An equivalent example to the <InlineCode>routes</InlineCode> example above, but instead using{' '}
```

  <InlineCode>continue</InlineCode> to rewrite a URL without matching.
</Caption>

```json
{
  "headers": [
    {
      "source": "/test",
      "headers": [
        {
          "key": "Cache-Control",
          "value": "max-age=600"
        }
      ]
    }
  ],
  "rewrites": [
    { "source": "/(.*)", "destination": "/src/public/$1" },

>     { "source": "/src/public/test", "destination": "/src/function/test" }

Reviewer: I believe code examples should reflect production usage, so maybe something other than test
```suggestion
    { "source": "/src/public/get-users", "destination": "/src/function/get-users" }
```

Author: This was meant to match the implementation above it (which was using test).

Feel free to create a new PR to update both to something that reflects production

  ]
}
```

<Caption>

>   The equivalent to the routes example above using{' '}

Reviewer: ```suggestion
  An equivalent example to the <InlineCode>routes</InlineCode> example above, but instead using{' '}
```

  A correct example of a <InlineCode>now.json</InlineCode> route.
</Caption>

When using `rewrites`, the `.` character is considered a literal unless wrapped in a group.

```json
{
  "rewrites": [
    { "source": "/test/file.json", "destination": "/api/generate-json" },
    { "source": "/proxy/(.*)", "destination": "https://example.com/$1" }
  ]
}
```

<Caption>

>   The first rewrite is generating a json file from a serverless function, the

Reviewer: ```suggestion
  The first rewrite generates a <InlineCode>json</InlineCode> file from a Serverless Function. The
```

>   second rewrite is proxying all traffic to <InlineCode>example.com</InlineCode>.

Reviewer: ```suggestion
  second rewrite proxies all traffic to <InlineCode>example.com</InlineCode>.
```

  A <InlineCode>now.json</InlineCode> route using a catch-all except regex.
</Caption>

```json
{
  "rewrites": [{ "source": "/blog/:post", "destination": "/blog" }]
}
```

<Caption>

>   The equivalent to the routes example above using{' '}

Reviewer: ```suggestion
  An equivalent example to the <InlineCode>routes</InlineCode> example above, but instead using{' '}
```
